manager/integration/tests/test_infra.py

```diff
@@ -0,0 +1,69 @@
+import os
+import pytest
+import time
+
+from digitalocean import digitalocean
+from common import get_core_api_client, get_longhorn_api_client
+
+
+def detect_cloudprovider():
+    cloudprovider_name = os.getenv("CLOUDPROVIDER")
+
+    if cloudprovider_name == "digitalocean":
+        cloudprovider = digitalocean()
+        return cloudprovider
+
+
+@pytest.mark.infra
+def test_offline_node():
+    node_worker_label = 'node-role.kubernetes.io/worker'
+    pod_lable_selector = "longhorn-test=test-job"
+
+    k8s_api_client = get_core_api_client()
+    longhorn_api_client = get_longhorn_api_client()
+    cloudprovider = detect_cloudprovider()
+
+    for pod in k8s_api_client.list_namespaced_pod(
+                    'default',
+                    label_selector=pod_lable_selector).items:
+
+        if pod.metadata.name == "longhorn-test":
+            longhorn_test_node_ip = pod.status.host_ip
+
+    for node_item in k8s_api_client.list_node().items:
+        if node_worker_label in node_item.metadata.labels and \
+                node_item.metadata.labels[node_worker_label] == 'true':
+            node_name = node_item.metadata.name
+            node_external_ip = \
+                node_item.metadata.annotations['rke.cattle.io/external-ip']
+            if node_external_ip == longhorn_test_node_ip:
+                continue
+            else:
+                break
+
+    node = cloudprovider.node_id(node_external_ip)
+
+    cloudprovider.node_shutdown(node)
+
+    while True:
```

Comment:
We cannot wait in an infinite loop, and we should have sleep between retrying otherwise it can be too many API calls. We should have a retry interval and maximum retry count. See https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/common.py#L1114 for examples.